test/Ops-proxy.test.ts

```diff
@@ -128,6 +129,40 @@ describe("Ops Proxy module test", function () {
     expect(await opsProxy.owner()).to.be.eql(userAddress);
   });
 
+  it("proxy - only owner can update implementation", async () => {
+    const [proxyAddress] = await opsProxyFactory.getProxyOf(userAddress);
+    const opsProxy: EIP173ProxyWithCustomReceive = await ethers.getContractAt(
+      "EIP173ProxyWithCustomReceive",
+      proxyAddress
+    );
+
+    await opsProxy.connect(user).upgradeTo(ETH);
+    const implementationAddress = ethers.utils.hexStripZeros(
+      await ethers.provider.getStorageAt(
+        opsProxy.address,
+        "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"
+      )
+    );
+
+    expect(implementationAddress).to.be.eql(ETH);
+
+    await expect(
+      opsProxy.connect(user2).upgradeTo(ZERO_ADD)
```

Comment:
I think for security purpose, the upgradeTo method should verify that the upgradeAddress is somehow whitelisted on the proxyFactory?
Otherwise an attacker could target a user and try to make him update its proxy with a malicious implementation, and we have no control over that.